### ISSUE TITLE
FR: transparency support

### ISSUE THREAD

**Issue by noerw** [TobleMiner/shoreline#17]
The framebuffer does not apply transparency correctly. It seems alpha values other than 0 are ignored?

See this comparison between shoreline (top) and [pixelnuke](https://github.com/defnull/PixelNuke) (bottom) with similar inputs:

|||
|--|--|
|![Screenshot from 2021-01-01 05-11-23](https://user-images.githubusercontent.com/7880552/103433360-adc79980-4be7-11eb-8b07-90ecae6ffeef.png) | ![Screenshot from 2021-01-01 05-08-55](https://user-images.githubusercontent.com/7880552/103433376-f41cf880-4be7-11eb-8e2f-1cadecd04ccb.png)|

Thanks for this amazingly fast toy <3

**Comment by TobleMiner** [comment]
Yes, there is no full transparency support. However, at the time of implementation that was a contious decision to keep performance as high as possible. I could implement full transparency support as an optional feature, that needs to be enabled/disabled at compile time though. Would that be fine with you?

**Comment by noerw** [comment]
Yeah, that sounds perfect! I just looked at the code, it'll be interesting to see the performance difference

**Comment by TobleMiner** [comment]
Please have a look at https://github.com/TobleMiner/shoreline/tree/feature-alpha-blending

I'm not sure whether or not that is waht exactly you are looking for. Please let me know if it's ok or not.

**Comment by TobleMiner** [comment]
I've just updated the branch again. Now transparent colors can also be composed ontop of one another

**Comment by noerw** [comment]
Thanks, that was quick!

Looks good so far. Just a minor thing: There is some flickering going on now on some pixels, when no data is coming in: 
![Peek 2021-01-02 04-53](https://user-images.githubusercontent.com/7880552/103450453-8ee00a80-4cae-11eb-8d14-a768fb282364.gif)
First thought was that this might be because of `-Ofast` / `-ffast-math`, but compiling without this option gives the same results.
If you think it's related to specific input data, I can test with other inputs.
*edit:* or is this #2?

Regarding performance: a standard build with SDL frontend peaks at 6.2 Gb/s on my laptop. With ALPHA_BLENDING it peaks at 5.4 Gb/s., so ~12% perf is lost.

**Comment by TobleMiner** [comment]
Cool! The flickering is interesting. Right now I'm not entirely sure where it is coming from. Would you mind providing me with the picture you are using for testing? That could make replicating the issue a lot easier.

I'm pretty sure it is not #2, that should be strictly a performance thing. The issue you are seeing might also be related to the coalescing mechanism though.

**Comment by noerw** [comment]
to reproduce:
```sh
git clone https://github.com/speckij/hochwasser -b new-rgbsplit
cd hochwasser
go build .
wget https://nroo.de/pixelflut/pro_aktion.png
(sleep 0.2; echo 'rgbsplit' ) | ./hochwasser -image pro_aktion.png -order shuffle
```

**Comment by TobleMiner** [comment]
Sorry for the delay. 5a0bcb966801c690d6f6852bc08f7f4bb9d1c3da should address the flickering you were seeing

**Comment by noerw** [comment]
Just came back to this; your last commit fixed the flickering :) Thanks
